src/lua/datetime.lua

```diff
@@ -145,6 +250,22 @@ local function check_range(v, from, to, txt, extra)
     end
 end
 
+local function check_dt(dt_v, direction, v, unit)
+    if dt_v >= INT_MIN and dt_v <= INT_MAX then
+        return
+    end
+    local operation = direction >= 0 and 'addition' or 'subtraction'
+    local title = unit ~= nil and ('%s of %d %s'):format(operation, v, unit) or
+                                  operation
```

Comment:
The fast track is in 1st 3 lines of a `check_dt` function - we bail out of function if everything is correct.

https://github.com/tarantool/tarantool/blob/a36b65f41152c860b82c4088cc096bcf8a398067/src/lua/datetime.lua#L254-L256

Here we are in the error case, which is requiring error raising, so these format string preparations are needed

Reply:
Ok, I was confused by the `if` branches having a not covered `else` part.
```Lua
local function check_dt(dt_v, direction, v, unit)
    if dt_v >= INT_MIN and dt_v <= INT_MAX then
        return
    end
    local operation = direction >= 0 and 'addition' or 'subtraction'
    local title = unit ~= nil and ('%s of %d %s'):format(operation, v, unit) or
                                  operation
    if dt_v < INT_MIN then
        error(('%s makes date less than minimum allowed %s'):
                format(title, MIN_DATE_TEXT), 3)
    elseif dt_v > INT_MAX then
        error(('%s makes date greater than maximum allowed %s'):
                format(title, MAX_DATE_TEXT), 3)
    end
end
```
Here in the end you check `dt_v < INT_MIN`, `dt_v > INT_MAX`, and the value inside of the range looks not covered. Even though it is not true, because you already did an early return above. I suggest to either add `assert(false)` into `else` or turn one of the branches into `else` and assert its condition.

Reply:
Added `assert` for final `else` branch